docs/method-monitoring.asciidoc

```diff
@@ -0,0 +1,258 @@
+} catch (Exception e) {
+    span.captureException(e);
+    throw e;
+} finally {
+    span.end();
+}
+----
+<1> Get current span
+<2> Create a child span
+<3> Make this span the active span on the current thread
+<4> Override the default span name
+<5> Add labels to the span
+
+[float]
+==== Combine with annotations
+
+// This section is sourced from the <<method-annotations>> documentation
+include::method-monitoring.asciidoc[tag=combine-api-annotations]
+
+//********************************************
+[[method-annotations]]
+=== Annotations
+
+The <<api-annotation,annotation API>> allows you to place annotations on top of methods to automatically create spans for them.
+This method of creating spans is easier, more robust, and typically more performant than using the API;
+there’s nothing you can do wrong like forgetting to end a span or close a scope.
+
+Annotations are less flexible when used on their own, but can be combined with the span API for added flexibility.
+
+[float]
+==== How-to create spans with the annotations API
+
+Here's an example that uses the <<api-capture-span,`@CaptureSpan`>> annotation to create a span for the `spanWithAnnotation()` method.
+The span is named `spanName`, is of type `ext`, and subtype `http`.
+
+[source,java]
+----
+@CaptureSpan(value = "spanName", type = "ext", subtype = "http")
+private static void spanWithAnnotation() {
+    // do your thing...
+}
+----
+
+[float]
+==== Combine with the span API
+
+// This content is also used in the `method-api` documentation
+// Ensure any changes made here can also apply there.
+// tag::combine-api-annotations[]
+You can combine annotations with the span API to increase their flexibility.
+Just get the current span on an annotated method and customize the span to your liking.
+
+[source,java]
+----
+@CaptureSpan <1>
+private static void spanWithAnnotation(String foo) {
+    Span span = ElasticApm.currentSpan(); <2>
+    span.setTag("foo", foo); <3>
+}
+----
+<1> Use `@CaptureSpan` annotation to create a span
+<2> Get the current span (the one created via the `@CaptureSpan` annotation)
+<3> Customize the span
+// end::combine-api-annotations[]
+
+//********************************************
+[[method-config-based]]
+=== Configuration-based
+
+Use the <<config-trace-methods,`trace_methods`>> configuration option to specify additional methods to instrument.
+You can match methods via wildcards in the package, class or method name, by their modifier (like public),
+by a particular annotation, and more.
+Because you don’t need to modify your source code, this makes it possible to monitor code in 3rd party libraries.
+
+Be careful, it's easy to overuse `trace_methods` by matching too many methods--hurting both runtime and startup performance.
+Use in conjunction with <<config-trace-methods-duration-threshold,`trace_methods_duration_threshold`>> when setting for entire packages
+in order to reduce overhead and avoid having too many spans in the APM app.
```

Comment:
It doesn't improve performance a lot so better not mention it to not give a false sense of security
```suggestion
in order to avoid having too many spans in the APM app.
```